server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -135,6 +140,11 @@ public ExtensionsManager(Path extensionsPath) throws IOException {
         // will be initialized in initializeServicesAndRestHandler which is called after the Node is initialized
         this.transportService = null;
         this.clusterService = null;
+        if (additionalSettings == null) {
```

Comment:
Super nit, with the goal to make immutable copy (of the original collection):
```
        this.additionalSettings = new HashSet<>();
        if (additionalSettings != null) {
            this.additionalSettings.addAll(additionalSettings);
        }
```

Reply:
done